src/gui/touchscreengui.cpp

```diff
@@ -450,67 +423,56 @@ TouchScreenGUI::TouchScreenGUI(IrrlichtDevice *device, IEventReceiver *receiver)
 	m_settingsbar(device, receiver),
 	m_rarecontrolsbar(device, receiver)
 {
-	for (unsigned int i=0; i < after_last_element_id; i++) {
-		m_buttons[i].guibutton     =  0;
-		m_buttons[i].repeatcounter = -1;
-		m_buttons[i].repeatdelay   = BUTTON_REPEAT_DELAY;
+	for (auto & m_button : m_buttons) {
+		m_button.guibutton     = nullptr;
+		m_button.repeatcounter = -1;
+		m_button.repeatdelay   = BUTTON_REPEAT_DELAY;
 	}
 
 	m_touchscreen_threshold = g_settings->getU16("touchscreen_threshold");
 	m_fixed_joystick = g_settings->getBool("fixed_virtual_joystick");
 	m_joystick_triggers_special1 = g_settings->getBool("virtual_joystick_triggers_aux");
 	m_screensize = m_device->getVideoDriver()->getScreenSize();
+	button_size = MYMIN(m_screensize.Y / 4.5f,
+						porting::getDisplayDensity() *
```

Comment:
Too many indents. If this is requested by clang format, please add `// clang-format off`